NitroxClient/Debuggers/SceneExtraDebugger.cs

```diff
@@ -76,27 +81,73 @@ private void RenderTabTools()
         {
             if (gameObjectResults.Count > 0)
             {
-                using GUILayout.ScrollViewScope scroll = new(hierarchyScrollPos);
-                hierarchyScrollPos = scroll.scrollPosition;
+                GUILayout.Label($" Found {gameObjectResults.Count} results.");
 
-                for (int index = 0; index < gameObjectResults.Count; index++)
+                using (GUILayout.ScrollViewScope scroll = new(hierarchyScrollPos))
                 {
-                    if (index > 30)
+                    hierarchyScrollPos = scroll.scrollPosition;
+
+                    int startIndex = resultsPerPage * searchPage;
+                    for (int index = startIndex; index < gameObjectResults.Count; index++)
                     {
-                        GUILayout.Label($"There are {gameObjectResults.Count - index} more results which aren't displayed", "fillMessage");
-                        break;
+                        if (index > startIndex + resultsPerPage)
```

Comment:
I would swap the condition of the for loop and the if. The for is counting from startIndex to resultsPerPage while the if should break the loop before the array access if we are out of bounds.